sw/device/lib/crypto/drivers/kmac_test.c

```diff
@@ -76,42 +80,105 @@ bool test_main(void) {
   // Below we repeat some of the test above, but through higher-level
   // cryptolib API. The gist is to test if glueing is done correctly.
   for (i = 0; i < nist_kmac_nr_of_vectors; i++) {
+    LOG_INFO("Testing cryptolib API (glueing) #%d", i);
     current_test_vector = nist_kmac_vectors[i];
 
-    crypto_const_uint8_buf_t input_buf;
-    input_buf.data = current_test_vector->input_str;
-    input_buf.len = current_test_vector->input_str_len;
+    hash_mode_t hash_mode;
+    xof_mode_t xof_mode;
+    mac_mode_t mac_mode;
 
-    crypto_uint8_buf_t digest_buf;
-    digest_buf.data = digest;
-    digest_buf.len = current_test_vector->digest_len;
+    // Max key size is 512 bits
+    uint32_t keyblob[2 * 512 / 32];
 
-    if (current_test_vector->operation == kKmacOperationSHA3) {
-      LOG_INFO("Testing cryptolib API (glueing) #%d", i);
-      hash_mode_t hash_mode;
-      switch (current_test_vector->security_str) {
-        case kKmacSecurityStrength224:
-          hash_mode = kHashModeSha3_224;
-          break;
-        case kKmacSecurityStrength256:
-          hash_mode = kHashModeSha3_256;
-          break;
-        case kKmacSecurityStrength384:
-          hash_mode = kHashModeSha3_384;
-          break;
-        case kKmacSecurityStrength512:
-          hash_mode = kHashModeSha3_512;
-          break;
-        default:
-          return false;
-      }
-
-      crypto_status_t err_status =
-          otcrypto_hash(input_buf, hash_mode, &digest_buf);
-      CHECK(err_status == kCryptoStatusOK);
-      CHECK_ARRAYS_EQ(current_test_vector->digest, digest_buf.data,
-                      current_test_vector->digest_len);
+    for (size_t j = 0; j < 2 * current_test_vector->key_len; j++) {
+      keyblob[j] = j < current_test_vector->key_len
+                       ? read_32(current_test_vector->key + 4 * j)
```

Comment:
nit: use `sizeof` rather than a hardcoded 4 so it's clear where the number comes from.
```suggestion
                       ? read_32(current_test_vector->key + sizeof(uint32_t) * j)
```